crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1846,7 +1852,13 @@ impl<'db> TypeInferenceBuilder<'db> {
 
                 unpacked.get(name_ast_id).unwrap_or(Type::Unknown)
             }
-            TargetKind::Name => value_ty,
+            TargetKind::Name => {
+                if self.file().is_stub(self.db().upcast()) && value.is_ellipsis_literal_expr() {
```

Comment:
Nit: It might be worth introducing a `is_stub` function on `InferenceBuilder` that does the `self.file().is_stub(self.db().upcast())`